string  version_string
int64   maj
int64   min
int64   rev

string  product_name
uint32  product_id
string  desc
